Updates to pin numbers and sensors used

abill001 opened the pull request:
- Removed force plate code, replaced with different calibration factor for FX1901 50lbf load cell.
- Added Analog Temperature sensor code
- MK.2 Only sensors are : 3 chamber thermocouples
- Added tare for second pressure sensor
- TODO: error checking for analog temp data or load cell force? (this is a question I don't know the answer to)

abill001 committed:
Updates to pin numbers and sensors used

- Removed force plate code, replaced with different calibration factor for FX1901 50lbf load cell.
- Added Analog Temperature sensor code
- MK.2 Only sensors are : 3 chamber thermocouples
- Added tare for second pressure sensor
- TODO: error checking for analog temp data or load cell force? (this is a question I don't know the answer to)

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -19,80 +19,74 @@
-int pressure_val_num = 0;
-bool pressure_zero_ready = false;
-float pressure_zero_val = 0;
+float pressure_hist_vals_1[PRESSURE_NUM_HIST_VALS];
```
Since we have multiple pressure sensors, maybe turn these into 1D/2D arrays of size 2?  Would make things a bit cleaner.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -19,80 +19,74 @@
-HX711 scale(LOAD_CELL_DOUT, LOAD_CELL_CLK);
-
-// Force plate setup
-#define FORCE_PLATE_PIN A2
```
How absolutely sure are we that we will never use the force plate again for any purpose (e.g. demos like at the STEM fair last year?)  If so we should at least leave in the force plate code, but disable it via the preprocessor (e.g. make it part of the `DEMO` mode?)  If we know we will never use it again for sure though then it is fine to take it out.

abill001 replied:
We could use the load cell for those kinds of purpose. I'm 90% confident the load cell should work now that we're just plugging the breakout board into the pcb instead of soldering the physical amplifier. Ideally, this should be the direction we are going.

krame505 replied:
OK, sounds good then.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -19,80 +19,74 @@
 
 // Sensor data
-float chamber_temp, inlet_temp, outlet_temp, pressure, force, x,y,z;
+float chamber_temp_1, chamber_temp_2, chamber_temp_3, pressure_fuel, pressure_ox, force, x,y,z, inlet_temp, outlet_temp;
```
Let's make `chamber_temp_...` into an array.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -19,80 +19,74 @@
-int inlet_temp_error = 0;
-int outlet_temp_error = 0;
-int pressure_error = 0;
+int chamber_temp_1_error = 0;
```
Let's make `chamber_temp_..._error` into an array.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -169,44 +164,45 @@ void setup() {
   // Grab pressure data
-  pressure = (analogRead(PRESSURE_PIN) * 5 / 1024.0) * PRESSURE_CALIBRATION_FACTOR - (PRESSURE_OFFSET + pressure_zero_val); // Pressure is measured in PSIG
+  pressure_fuel = (analogRead(PRESSURE_FUEL) * 5 / 1024.0) * PRESSURE_CALIBRATION_FACTOR - (PRESSURE_OFFSET + pressure_zero_val_1); // Pressure is measured in PSIG
+  pressure_fuel = (analogRead(PRESSURE_OX) * 5 / 1024.0) * PRESSURE_CALIBRATION_FACTOR - (PRESSURE_OFFSET + pressure_zero_val_2); // Pressure is measured in PSIG
```
Pretty sure this should be `fuel_ox = ...`

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +234,40 @@ void loop() {
+    if (pressure_zero_ready_1) {
+      Serial.println(F("Zeroing fuel pressure"));
+      pressure_zero_ready_1 = false;
+      pressure_zero_val_1 = 0;
```
Since this code is now being duplicated you can probably factor it out into a function, at least partially.  Maybe write a function that computes the mean value of an array of floats and use that here instead?

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +234,40 @@ void loop() {
+      Serial.println(F("Zeroing oxidizer pressure"));
+      pressure_zero_ready_2 = false;
+      pressure_zero_val_2 = 0;
       for (int i = 0; i < PRESSURE_NUM_HIST_VALS; i++)
```
Same as above comment.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +234,40 @@ void loop() {
   }
-  #endif
+  
+  //TODO: Tare for temperature?
```
We probably can't (or at least don't want to) do this, since we are roughly interested in the actual temperatures here, and don't really have a point against which to zero the sensors.

krame505 reviewed (changes requested):
Looks pretty good, just a few minor-ish things that you should think about changing.

abill001 committed:
Tweaks to clean up code and fix typos

- Implemented pressure error data in arrays and incorporated functions
- Switch thermo data and error to arrays
- Still think we should get rid of force plate

abill001 committed:
Fixed SEND_FIELD for thermos

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -169,44 +159,33 @@ void setup() {
-    pressure_zero_ready = true;
-  }
-  #endif
+  tare_pressure (PRESSURE_FUEL);
```
I don't see this method defined anywhere - is this in another file that you forgot to commit?

krame505 replied:
Oh nvm, I see you opened a seperate pull request.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -19,80 +19,69 @@
+//Pressure sensor 0 = Fuel, Pressure sensor 1 = Oxidizer
+float pressure_hist_vals [NUMBER_OF_PRESSURE_SENSORS][PRESSURE_NUM_HIST_VALS];
+int pressure_val_num [NUMBER_OF_PRESSURE_SENSORS] = {0,0};
+bool pressure_zero_ready [NUMBER_OF_PRESSURE_SENSORS] = {false, false};
```
After thinking about this again, these 2 arrays don't actually need to be arrays - since we are just tracking when a sufficient number of data points have been received, they will all have the same values.  

With this change, this should simplify the tare code somewhat, and avoid the need for a seperate `tare_pressure` function.

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +217,18 @@ void loop() {
+  
+  //TODO: Update flag names to match the following code
+  READ_FLAG(zero_pressure_fuel) {
+    zero_pressure (PRESSURE_FUEL, "fuel");
```
Moving all this code into a seperate function doesn't actually simplify things that much - an important goal in software design is to maximize code reusability - having a function that takes in a pin number, uses that to find an array index, and then updates some global arrays based on that isn't very flexible or obvious what it does.  Duplicating code is sometimes OK, and often preferable when dealing with global variables.  

Instead, consider writing a function `float mean(unsigned count, float data[])` that computes the mean of a `float` array, and using that to simplify the existing zeroing code.  You can pass in the values of global variables, and use the result to update the appropriate global value.

abill001 committed:
update to tare pressure and zero pressure code

abill001 committed:
Added mean function

abill001 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +222,32 @@ void loop() {
-  #if CONFIGURATION == MK_2
-  READ_FLAG(zero_pressure) {
+  
+  //TODO: Update flag names to match the following code 
```
Lucas you never answered this question, does this need to be done at all or is the flag naming correct?

krame505 commented on static_test_driver/sensors.ino:
```diff
@@ -1,3 +1,9 @@
+  float result;
+  for (int i = 0; i < size; i++)
+      result += data[i];
+  return result /= size;
```
Quick style thing: No need to assign to result, you can just `return result / size;`

krame505 commented on static_test_driver/static_test_driver.ino:
```diff
@@ -235,40 +222,32 @@ void loop() {
-      pressure_zero_val /= PRESSURE_NUM_HIST_VALS;
+      Serial.print(F("Zeroing fuel pressure"));
+      pressure_zero_val[0] = 0;
+      mean(&pressure_hist_vals[0][0], PRESSURE_NUM_HIST_VALS);
```
This line doesn't do anything with the value returned by `mean`.  Also `&foo[x][0]` can be rewritten as just `foo[x]`, so these 2 lines should be just
`pressure_zero_val[0] = mean(pressure_hist_vals[0], PRESSURE_NUM_HIST_VALS);`

Same for below.

krame505 commented:
The flag names can be whatever you want, we just need to change them to match in the telemetry interface code.

abill001 committed:
Fixed hurried code error for pressure_zero_val

abill001 committed:
style for mean function

krame505 commented on static_test_driver/sensors.ino:
```diff
@@ -2,7 +2,7 @@ float mean (const float *data, unsigned int size) {
   float result;
```
Oh, one more thing I forgot: You shouldn't leave result uninitialized.

abill001 replied:
Lol, yeah that would have been bad. Sorry about that.

abill001 committed:
fixed hurried code error ... again
